### ISSUE TITLE
Monitor actor code cleanup

### ISSUE THREAD

**Issue by da-kami** [itchysats/itchysats#1773]
When working on https://github.com/itchysats/itchysats/pull/1765 I noticed that the monitor actor has some duplicated code parts and it looks like several code parts could be simplified and optimized.

In particular these questions popped up: 

- We are storing `Option<Transaction` for cet / collab-settlement-tx (...) for re-publication of transactions, but keep a separate `Option<(Txid, Script>)` for monitoring. Is that split necessary? Wouldn't it be enough to have logic like "If tx is `Some` then we have to republish *and* monitor it.
- Make a more generalized "extract `Txid` and `Script`" for montitoring.
- Sometimes it is hard to follow the code-pathes in the monitor actor.
- Once #320 is done we have to ensure to also try re-publishing the collab-settlement transaction in case it failed before. At the moment this is never re-tried, which could lead to (the unlikely case of) being stuck in collab-settlement without the transaction being published.

**Comment by bonomat** [comment]
This project is unmaintained
